pkg/i2gw/provider.go

```diff
@@ -68,13 +70,19 @@ type CustomResourceReader interface {
 	ReadResourcesFromFile(ctx context.Context, filename string) error
 }
 
-// The ResourceConverter interface specifies all the implemented Gateway API resource
-// conversion functions.
+// The ResourceConverter interface specifies conversion functions from Ingress
+// and extensions into IR.
 type ResourceConverter interface {
+	// ToIR converts stored API entities associated with the Provider into IR.
+	ToIR() (IR, field.ErrorList)
+}
 
-	// ToGatewayAPIResources converts stored API entities associated
-	// with the Provider into GatewayResources.
-	ToGatewayAPI() (GatewayResources, field.ErrorList)
+// The ResourceConverter interface specifies conversion functions from IR
+// into Gateway and Gateway extensions.
+type ExtensionConverter interface {
```

Comment:
Am I right that the purpose of this interface is to implement transition from the IR to Gateway API resources and extensions? If yes, `ExtensionConverter` name is a bit confusing. Maybe `IRToGatewayAPIConverter`?

I'd like to hear more thoughts. 

cc: @arkodg @robscott

Reply:
similarly, `ResourceConverter` can be changed to `ResourcesToIR` or something similar

Reply:
Done.